lavalink/player.py

```diff
@@ -119,6 +119,47 @@ def connected(self) -> bool:
         """
         return self._connected
 
+    async def on_voice_server_update(self, data: dict) -> None:
+        self._pending_server_update = data
+        await self._send_lavalink_voice_update()
+
+    async def on_voice_state_update(self, data: dict) -> None:
+        self._session_id = data["session_id"]
+        if (channel_id := data["channel_id"]) is None:
+            ws_rll_log.info("Received voice disconnect from discord, removing player.")
+            ws_rll_log.verbose("Voice disconnect from discord: %s -  %r", data, self)
+            self._session_id = None
+            self._pending_server_update = None
+            await self.disconnect(force=True)
+        else:
+            channel = self.guild.get_channel(int(channel_id))
+            if channel != self.channel:
+                if self.channel:
+                    self._last_channel_id = self.channel.id
+                elif channel is None:
```

Comment:
```suggestion
                if channel is None:
```
because when `self.channel is not None` and `channel is None` are both true, this wouldn't otherwise trigger.